expr/normalize.go

```diff
@@ -0,0 +1,90 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/consolidation"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/util"
+)
+
+// normalize normalizes series to the same common LCM interval - if they don't already have the same interval
+// any adjusted series gets created in a series drawn out of the pool and is added to the cache so it can be reclaimed
+func normalize(cache map[Req][]models.Series, in []models.Series) []models.Series {
+	var intervals []uint32
+	for _, s := range in {
+		if s.Interval == 0 {
+			panic("illegal interval 0")
```

Comment:
Do we need to panic here or is there another way to handle this?

Reply:
panic is fine. it will just affect this one request